solvers/wscd.py

```diff
@@ -0,0 +1,183 @@
+from benchopt import BaseSolver, safe_import_context
+
+with safe_import_context() as import_ctx:
+    import numpy as np
+    from numba import njit
+
+
+if import_ctx.failed_import:
+
+    def njit(f):  # noqa: F811
+        return f
+
+
+@njit
+def st(x, mu):
+    if x > mu:
+        return x - mu
+    if x < -mu:
+        return x + mu
+    return 0
+
+
+@njit
+def prox_mcp(x, lmbd, gamma):
+    if gamma < 1:
+        return 0 if abs(x) <= lmbd else x
+        # TODO potential numerical errors in <=  leading to hard discontinuity
+
+    if x > gamma * lmbd:
+        return x
+    if x < -gamma * lmbd:
+        return x
+    return gamma / (gamma - 1) * st(x, lmbd)
+
+
+@njit
+def norm_col_sparse(data, indptr):
+    n_features = len(indptr) - 1
+    norms = np.zeros(n_features)
+    for j in range(n_features):
+        tmp = 0
+        for idx in range(indptr[j], indptr[j + 1]):
+            tmp += data[idx] ** 2
+        norms[j] = np.sqrt(tmp)
+    return norms
+
+
+@njit
+def cd(X, y, lmbd, gamma, lipschitz, n_iter, winit=None, tol=1e-3):
```

Comment:
numba will not like `winit=None` and anyways it seems that it in not used in the function